tests/router/test_router_e2e_with_mockers.py

```diff
@@ -432,12 +381,6 @@ def test_mocker_two_kv_router(request, runtime_services, predownload_tokenizers)
             f"Successfully completed {NUM_REQUESTS} requests across {len(router_ports)} routers"
         )
 
-        # Check etcd registration - expect 2 KV routers
-        # Use the mockers' endpoint since all mockers share the same component path
-        asyncio.run(
-            check_registration_in_etcd(expected_count=2, endpoint=mockers.endpoint)
```

Comment:
@grahamking any recommendations on the new path forward to do a check like this? Do we have plans for some kind of more generic registry APIs to examine the components in the system?

FYI @PeaBrane on the change